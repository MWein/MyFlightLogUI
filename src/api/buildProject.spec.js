import {
  headerData,
  getEntriesFromPhase,
  createHoursGraphObject,
  createExpensesGraphObject,
} from './buildProject'

describe('getEntriesFromPhase', () => {
  const mockBuildData = {
    phases: [
      {
        id: 'r',
        entries: ['Research1', 'Research2', 'Research3'],
      },
      {
        id: 'w',
        entries: ['Workshop1', 'Workshop2'],
      },
    ],
  }

  it('Returns all entries if selectedPhase is all', () => {
    expect(getEntriesFromPhase(mockBuildData, 'all')).toMatchSnapshot()
  })

  it('Returns only entries belonging to a particular phase', () => {
    expect(getEntriesFromPhase(mockBuildData, 'w')).toMatchSnapshot()
  })
})
