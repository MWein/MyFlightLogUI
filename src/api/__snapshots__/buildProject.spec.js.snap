// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEntriesFromPhase Returns all entries if selectedPhase is all 1`] = `
Array [
  "Research1",
  "Research2",
  "Research3",
  "Workshop1",
  "Workshop2",
]
`;

exports[`getEntriesFromPhase Returns only entries belonging to a particular phase 1`] = `
Array [
  "Workshop1",
  "Workshop2",
]
`;
